### PR TITLE
pkg/metastore: Reduce allocs in MakeLocationID

### PR DESCRIPTION
**Pull request by marselester**
Reduced allocations in `MakeLocationID`.

```
$ go test -run=^$ -bench ^BenchmarkMakeLocationID$ -benchmem ./pkg/metastore -count=1
...
# new
BenchmarkMakeLocationID-12    	 3224959	       370.4 ns/op	      96 B/op	       1 allocs/op
# old
BenchmarkMakeLocationID-12    	 1859330	       647.9 ns/op	     560 B/op	      10 allocs/op
```

This helped to shave 17K allocations per operation in `NormalizeWriteRawRequest`.

```
$ benchstat bench-old.txt bench-new.txt
name                         old time/op    new time/op    delta
NormalizeWriteRawRequest-12    43.9ms ± 7%    43.1ms ± 5%  -1.89%  (p=0.001 n=47+46)

name                         old alloc/op   new alloc/op   delta
NormalizeWriteRawRequest-12    53.5MB ± 1%    52.7MB ± 1%  -1.62%  (p=0.000 n=50+50)

name                         old allocs/op  new allocs/op  delta
NormalizeWriteRawRequest-12      412k ± 0%      395k ± 0%  -4.12%  (p=0.000 n=50+50)
```

Together with https://github.com/parca-dev/parca/pull/3061 it should save ~31K/operation.

### CODE REVIEW

File: pkg/metastore/kv.go
Comment (marselester) [inline pkg/metastore/kv.go:73]: Note, I have dropped this logic because `binary.Write(hash, binary.BigEndian, 1)` and `binary.Write(hash, binary.BigEndian, 0)` produce an empty slice https://go.dev/play/p/4vl0F7x7o5n. Please let me know if that's a bug.

```go
package main

import (
	"bytes"
	"encoding/binary"
	"fmt"
)

func main() {
	b := &bytes.Buffer{}
	binary.Write(b, binary.BigEndian, 1)
	binary.Write(b, binary.BigEndian, 0)

	fmt.Println(b.Bytes())
	// Output:
	// []
}
```

File: pkg/metastore/kv_test.go
Comment (marselester) [inline pkg/metastore/kv_test.go:38]: I grabbed these values (added print statements and ran the tests) from `MakeLocationID` before making changes to it.

File: pkg/metastore/badger.go
Comment (kakkoyun) [inline pkg/metastore/badger.go:34]: The_Keymaker

**Review by kakkoyun — APPROVED** [review]
LGTM.

I'll wait for another set of eyes on it before merging it.

**Comment by marselester** [comment]
Thank you @kakkoyun! There is another PR coming that touches `kv.go`. I will hold it until this one is merged.

**Comment by kakkoyun** [comment]
@marselester Besides the unit tests and benchmarks, could you confirm everything works as expected with a smoke test? (actually, running Parca with the agent and ensuring everything works visually) Sorry for the toil, we don't have a fully-fledged integration test for these parts.

**Comment by marselester** [comment]
Symbolization looks good. In Parca logs I can see `level=error name=parca ts=2023-05-16T13:45:31.074463419Z caller=symbolizer.go:315 component=symbolizer msg="failed to remove liner file" err="remove /tmp/parca-symbolizer-3510594462: no such file or directory"`. The error appears when I run Parca built from the main branch as well, so I don't think this PR introduced a regression.

Screenshot 2023-05-16 at 09 43 47

<details>

<summary>Parca logs</summary>

```
$ ./parca 
ooooooooo.
`888   `Y88.
 888   .d88'  .oooo.   oooo d8b  .ooooo.   .oooo.
 888ooo88P'  `P  )88b  `888""8P d88' `"Y8 `P  )88b
 888          .oP"888   888     888        .oP"888
 888         d8(  888   888     888   .o8 d8(  888
o888o        `Y888""8o d888b    `Y8bod8P' `Y888""8o



level=info name=parca ts=2023-05-16T13:41:21.008669667Z caller=factory.go:52 msg="loading bucket configuration"
level=info name=parca ts=2023-05-16T13:41:21.020607027Z caller=badger.go:54 msg="Set nextTxnTs to 0"
level=info name=parca ts=2023-05-16T13:41:21.037130959Z caller=server.go:93 msg="starting server" addr=:7070
level=error name=parca ts=2023-05-16T13:42:41.072758222Z caller=symbolizer.go:315 component=symbolizer msg="failed to remove liner file" err="remove /tmp/parca-symbolizer-318771116: no such file or directory"
level=error name=parca ts=2023-05-16T13:43:11.072779186Z caller=symbolizer.go:315 component=symbolizer msg="failed to remove liner file" err="remove /tmp/parca-symbolizer-2220210787: no such file or directory"
level=error name=parca ts=2023-05-16T13:43:41.070749192Z caller=symbolizer.go:315 component=symbolizer msg="failed to remove liner file" err="remove /tmp/parca-symbolizer-1588673323: no such file or directory"
level=error name=parca ts=2023-05-16T13:44:21.095045717Z caller=symbolizer.go:315 component=symbolizer msg="failed to remove liner file" err="remove /tmp/parca-symbolizer-2726195421: no such file or directory"
level=error name=parca ts=2023-05-16T13:44:31.399561829Z caller=symbolizer.go:315 component=symbolizer msg="failed to remove liner file" err="remove /tmp/parca-symbolizer-701494168: no such file or directory"
level=error name=parca ts=2023-05-16T13:44:41.095659591Z caller=symbolizer.go:315 component=symbolizer msg="failed to remove liner file" err="remove /tmp/parca-symbolizer-2560791777: no such file or directory"
level=error name=parca ts=2023-05-16T13:45:21.074199524Z caller=symbolizer.go:315 component=symbolizer msg="failed to remove liner file" err="remove /tmp/parca-symbolizer-3375163472: no such file or directory"
level=error name=parca ts=2023-05-16T13:45:21.074251489Z caller=symbolizer.go:315 component=symbolizer msg="failed to remove liner file" err="remove /tmp/parca-symbolizer-3972392873: no such file or directory"
level=error name=parca ts=2023-05-16T13:45:31.074463419Z caller=symbolizer.go:315 component=symbolizer msg="failed to remove liner file" err="remove /tmp/parca-symbolizer-3510594462: no such file or directory"
```

</details>

<details>

<summary>Parca Agent logs</summary>

```
$ sudo ./dist/parca-agent --remote-store-address=:7070 --remote-store-insecure
ooooooooo.                                                  .o.                                            .
`888   `Y88.                                               .888.                                         .o8
 888   .d88'  .oooo.   oooo d8b  .ooooo.   .oooo.         .8"888.      .oooooooo  .ooooo.  ooo. .oo.   .o888oo
 888ooo88P'  `P  )88b  `888""8P d88' `"Y8 `P  )88b       .8' `888.    888' `88b  d88' `88b `888P"Y88b    888
 888          .oP"888   888     888        .oP"888      .88ooo8888.   888   888  888ooo888  888   888    888
 888         d8(  888   888     888   .o8 d8(  888     .8'     `888.  `88bod8P'  888    .o  888   888    888 .
o888o        `Y888""8o d888b    `Y8bod8P' `Y888""8o   o88o     o8888o `8oooooo.  `Y8bod8P' o888o o888o   "888"
                                                                      d"     YD
                                                                      "Y88888P'

level=info name=parca-agent ts=2023-05-16T13:41:24.480928836Z caller=main.go:292 msg="maxprocs: Leaving GOMAXPROCS=2: CPU quota undefined"
level=info name=parca-agent ts=2023-05-16T13:41:24.490873457Z caller=main.go:334 msg="eBPF is supported and enabled by the host kernel"
name=parca-agent ts=2023-05-16T13:41:24.491659047Z caller=main.go:396 msg=starting... node=ubuntu-jammy store=:7070
level=info name=parca-agent ts=2023-05-16T13:41:24.492028722Z caller=main.go:499 msg=rlimit cur=1048576 max=1048575
level=info name=parca-agent ts=2023-05-16T13:41:24.505079281Z caller=cpu.go:224 msg="Attempting to create unwind shards" count=50
level=info name=parca-agent ts=2023-05-16T13:44:48.186046879Z caller=maps.go:1043 component=bpf_maps msg="creating a new shard to avoid splitting the unwind table for a function"
```

</details>

**Comment by kakkoyun** [comment]
@marselester Thanks for the tests 💯

**Pull request merged**
